### PR TITLE
Added SPIFFE/SPIRE data in the tables for landscape

### LINKED ISSUE #1020: [Landscape] Add SPIFFEE Support to the Service Mesh Landscape
**Current State:**
SPIFFEE is a well-used specification amongst service meshes and their proxies, like Istio, Envoy, Kuma, NGINX Service Mesh,...

**Desired State:**
It would be helpful to individuals reviewing the landscape if the Functional section included a column for SPIFFEE Support. 

---
**Contributor Resources**
- [Instructions for updating the landscape](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#updating-the-service-mesh-landscape)Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).

### PR DESCRIPTION
**Pull request by Nikhil-Ladha**
Signed-off-by: Nikhil-Ladha <nikhilladha1999@gmail.com>

**Description**
- Added SPIFFE/SPIRE data for a few service meshes
- Added icons instead of 'Yes'/'No' in the table
- Fixed a build error because of the _news/2020-10-29-kubernetes-evolves-into-an-enterprise-platform.md file

This PR fixes #1020  

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@ajessup, it's nice to have your collaboration on this. Here's an initial definition to get us started. Your review / updates on these would be fantastic - https://github.com/layer5io/layer5/pull/1082/files#diff-554e4bbed019a67937e16e13b7a00facf375e42ba0fbdb1e9182ad6002c928dd

// @yogi4

**Comment by Nikhil-Ladha** [comment]
@leecalcote , just curious if by going through the preview it seems fine to you to go by this `?` or shall I update it with the other one we discussed on the slack?
Though I shall warn you it's just not a dashed circled question mark it also blinks with an `!` mark alternatively.

**Comment by leecalcote** [comment]
@Nikhil-Ladha, ha! I wasn't aware of the blinking effect. In that case, yes, let's stick with the version in this PR. :)

**Review by leecalcote — DISMISSED** [review]
Looks good to me. What do you think, @ajessup?

File: _data/landscape/meshes.yml [hunk 1]
```diff
@@ -70,6 +79,8 @@
   encryption: "Yes"
   multi-tenant-score: 1
   netdev-persona-score: 9
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:82]: SPIFFE support is partial here, since it's based on Istio

File: _data/landscape/meshes.yml [hunk 2]
```diff
@@ -70,6 +79,8 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:83]: AFAICT there's no SPIRE in Citrix Service Mesh

File: _data/landscape/meshes.yml [hunk 3]
```diff
@@ -108,6 +121,8 @@
   tracing: "OpenTracing compatible"
   encryption: "Yes"
   multi-tenant-score: 1
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:124]: Consul has partial SPIFFE support, but doesn't support the Workload API or Federation, and thus is not interoperable with other SPIFFE implementations.

File: _data/landscape/meshes.yml [hunk 4]
```diff
@@ -108,6 +121,8 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:125]: Consul does not bundle SPIRE

File: _data/landscape/meshes.yml [hunk 5]
```diff
@@ -123,6 +138,8 @@
   encryption: "Yes"
   multi-tenant-score: 5
   netdev-persona-score: 1
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:141]: Istio has partial SPIFFE support, but doesn't support the Workload API or Federation API currently, and thus is not interoperable with other SPIFFE implementations.

File: _data/landscape/meshes.yml [hunk 6]
```diff
@@ -123,6 +138,8 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:142]: Istio does not use SPIRE by default

File: _data/landscape/meshes.yml [hunk 7]
```diff
@@ -25,6 +28,8 @@
   tracing: "Yes"
   encryption: "Yes"
   multi-tenant-score: 6
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:31]: Not supported today, but in development https://github.com/aws/aws-app-mesh-roadmap/issues/68

File: _data/landscape/meshes.yml [hunk 8]
```diff
@@ -25,6 +28,8 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:32]: Not supported today, but in development https://github.com/aws/aws-app-mesh-roadmap/issues/68

File: _data/landscape/meshes.yml [hunk 9]
```diff
@@ -162,6 +181,8 @@
   encryption: "Yes"
   multi-tenant-score: 1
   netdev-persona-score: 5
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:184]: SPIFFE IDs are not supported in Linkerd

File: _data/landscape/meshes.yml [hunk 10]
```diff
@@ -162,6 +181,8 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:185]: No

File: _data/landscape/meshes.yml [hunk 11]
```diff
@@ -177,6 +198,8 @@
   encryption: "Yes"
   multi-tenant-score: 1
   netdev-persona-score: 5
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:201]: Not supported

File: _data/landscape/meshes.yml [hunk 12]
```diff
@@ -177,6 +198,8 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:202]: No

File: _data/landscape/meshes.yml [hunk 13]
```diff
@@ -201,6 +224,8 @@
   encryption: "Yes"
   multi-tenant-score: "?"
   netdev-persona-score: "?"
+  spiffe: "Yes"
+  spire: "No"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:228]: Yes. https://docs.nginx.com/nginx-service-mesh/tutorials/kic/deploy-with-kic/

File: _data/landscape/meshes.yml [hunk 14]
```diff
@@ -225,6 +250,8 @@
   encryption: "No"
   multi-tenant-score: 1
   netdev-persona-score: 3
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:253]: No

File: _data/landscape/meshes.yml [hunk 15]
```diff
@@ -225,6 +250,8 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:254]: No

File: _data/landscape/meshes.yml [hunk 16]
```diff
@@ -248,6 +275,8 @@
   tracing: "Jaeger/Grafana"
   encryption: "Yes"
   multi-tenant-score: 10
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:278]: Partial, as with other SPIFFE implementations

File: _data/landscape/meshes.yml [hunk 17]
```diff
@@ -248,6 +275,8 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:279]: No

File: _data/landscape/meshes.yml [hunk 18]
```diff
@@ -262,6 +291,8 @@
   tracing: "?"
   encryption: "?"
   multi-tenant-score: 8
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:294]: No

File: _data/landscape/meshes.yml [hunk 19]
```diff
@@ -262,6 +291,8 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:295]: No

File: _data/landscape/meshes.yml [hunk 20]
```diff
@@ -283,10 +314,13 @@
   encryption: "?"
   multi-tenant-score: 8
   netdev-persona-score: 7
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:317]: Partial, as with other Istio implementations

File: _data/landscape/meshes.yml [hunk 21]
```diff
@@ -283,10 +314,13 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:318]: No

File: _data/landscape/meshes.yml [hunk 22]
```diff
@@ -297,10 +331,13 @@
   encryption: "Yes"
   multi-tenant-score: 10
   netdev-persona-score: 8
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:334]: No

File: _data/landscape/meshes.yml [hunk 23]
```diff
@@ -297,10 +331,13 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:335]: No

File: _data/landscape/meshes.yml [hunk 24]
```diff
@@ -311,6 +348,8 @@
   encryption: "Yes"
   multi-tenant-score: 1
   netdev-persona-score: 5
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:351]: No, but the community is exploring migrating to SPIFFE https://github.com/openservicemesh/osm/issues/1681

File: _data/landscape/meshes.yml [hunk 25]
```diff
@@ -311,6 +348,8 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:352]: Not currently

File: _data/landscape/meshes.yml [hunk 26]
```diff
@@ -333,11 +372,13 @@
   prometheus: "<i><a class='blue-grey-text' href='https://blog.turbinelabs.io/turbine-labs-is-shutting-down-and-our-team-is-joining-slack-2ad41554920c'>Project shutdown</a></i>"
   tracing: "<i><a class='blue-grey-text' href='https://blog.turbinelabs.io/turbine-labs-is-shutting-down-and-our-team-is-joining-slack-2ad41554920c'>Project shutdown</a></i>"
   encryption: "<i><a class='blue-grey-text' href='https://blog.turbinelabs.io/turbine-labs-is-shutting-down-and-our-team-is-joining-slack-2ad41554920c'>Project shutdown</a></i>"
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:375]: No

File: _data/landscape/meshes.yml [hunk 27]
```diff
@@ -333,11 +372,13 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:376]: No

File: _data/landscape/meshes.yml [hunk 28]
```diff
@@ -347,6 +388,8 @@
   tracing: "Pluggable"
   encryption: "Yes"
   multi-tenant-score: 1
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:391]: No

File: _data/landscape/meshes.yml [hunk 29]
```diff
@@ -347,6 +388,8 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:392]: No

File: _data/landscape/meshes.yml [hunk 30]
```diff
@@ -361,6 +404,8 @@
   tracing: "?"
   encryption: "?"
   multi-tenant-score: 1
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:407]: No

File: _data/landscape/meshes.yml [hunk 31]
```diff
@@ -361,6 +404,8 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:408]: No

File: _data/landscape/meshes.yml [hunk 32]
```diff
@@ -375,6 +420,8 @@
   tracing: "No"
   encryption: "Yes"
   multi-tenant-score: 1
+  spiffe: "?"
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:424]: No

File: _data/landscape/meshes.yml [hunk 33]
```diff
@@ -375,6 +420,8 @@
   tracing: "No"
   encryption: "Yes"
   multi-tenant-score: 1
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:423]: No

File: _data/landscape/meshes.yml [hunk 34]
```diff
@@ -389,6 +436,8 @@
   tracing: "?"
   encryption: "?"
   multi-tenant-score: 1
+  spiffe: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:439]: The project is working on exposing this externally https://www.youtube.com/watch?v=fxtG1wuLSDw&feature=youtu.be&t=17m37s

File: _data/landscape/meshes.yml [hunk 35]
```diff
@@ -389,6 +436,8 @@
...
+  spire: "?"
```
Comment (ajessup) [inline _data/landscape/meshes.yml:440]: Yes

**Review by ajessup — COMMENTED** [review]
This is a great start! I suggested some additional context in the comments.

**Comment by ajessup** [comment]
I'd suggest at some point in a future PR, rather than grade by "SPIFFE support" or "SPIRE support" the more meaningful model for developers will be "How much of the SPIFFE spec does it support", which right now is basically one of:
* None
* Partial (eg. Consul, Istio, Kuma)
* Complete (Nginx, Grey Matter, and anything else that uses SPIRE for its SPIFFE IDs)

Where *Complete* means the implementation supports the SPIFFE Workload API and SPIFFE Federation API.

**Comment by Nikhil-Ladha** [comment]
Thanks for the review @ajessup and the info for each of the mesh. I will update the data in the table with the provided values.

File: _data/landscape/meshes.yml (hunk 3 above)
Comment (Nikhil-Ladha) [reply _data/landscape/meshes.yml:124]: This is Grey Matter.

File: _data/landscape/meshes.yml (hunk 4 above)
Comment (Nikhil-Ladha) [reply _data/landscape/meshes.yml:125]: Same as above

**Review by leecalcote — APPROVED** [review]
Gents, this looks fantastic. Thank you, @Nikhil-Ladha and @ajessup. 👍

**Comment by ajessup** [comment]
Love it! Great work everyone!

**Comment by leecalcote** [comment]
💯 🎈

**Pull request merged**